javalanglib/src/main/scala/java/lang/Math.scala

```diff
@@ -219,5 +219,102 @@ object Math {
     }
   }
 
-  // TODO The methods not available in the JavaScript Math object
+  def addExact(a: scala.Int, b: scala.Int): scala.Int = {
+    val res = a + b
+    val resSgn = Integer.signum(res)
+    if (resSgn == Integer.signum(a) || resSgn == Integer.signum(b) ||
+        resSgn == 0 && a != Integer.MIN_VALUE && b != Integer.MIN_VALUE) res
+    else throw new ArithmeticException("Integer overflow")
+  }
+
+  def addExact(a: scala.Long, b: scala.Long): scala.Long = {
+    val res = a + b
+    val resSgn = Long.signum(res)
+    if (resSgn == Long.signum(a) || resSgn == Long.signum(b) ||
+        resSgn == 0 && a != Long.MIN_VALUE && b != Long.MIN_VALUE) res
+    else throw new ArithmeticException("Long overflow")
```

Comment:
Hum, it seems I messed up ^^ Good catch!

Then maybe consider this alternative formulation where we use only the *sign bit*, rather than the sign:
```scala
val resSgnBit = res < 0
if (resSgnBit == (a < 0) || resSgnBit == (b < 0)) res
else ...
```
On second thought, this should even be faster than my earlier proposal. (and it certainly *will* be faster if we special-case `someLong < 0` in the optimizer some day).

The same applies to `Int`, and also to `subtractExact`.